sdk/core/Azure.Core/src/Pipeline/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,341 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Copied from https://raw.githubusercontent.com/dotnet/corefx/master/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs
+
+using System;
+using System.Net.Http;
+using System.Net;
+using System.Collections.Generic;
+
+namespace System.Net.Http
+{
+    internal sealed class HttpEnvironmentProxyCredentials : ICredentials
+    {
+        // Wrapper class for cases when http and https has different authentication.
+        private readonly NetworkCredential _httpCred;
+        private readonly NetworkCredential _httpsCred;
+        private readonly Uri _httpProxy;
+        private readonly Uri _httpsProxy;
+
+        public HttpEnvironmentProxyCredentials(Uri httpProxy, NetworkCredential httpCred,
+                                                Uri httpsProxy, NetworkCredential httpsCred)
+        {
+            _httpCred = httpCred;
+            _httpsCred = httpsCred;
+            _httpProxy = httpProxy;
+            _httpsProxy = httpsProxy;
+        }
+
+        public NetworkCredential GetCredential(Uri uri, string authType)
```

Comment:
Is this interface implementation? If not, the authType parameter is not used.

Reply:
It is an implementation.